srtcore/api.cpp

```diff
@@ -2387,18 +2381,16 @@ void CUDTUnited::removeSocket(const SRTSOCKET u)
 
 void CUDTUnited::setError(CUDTException* e)
 {
-    delete (CUDTException*)pthread_getspecific(m_TLSError);
-    pthread_setspecific(m_TLSError, e);
+    SetThreadLocalError(e);
 }
 
-CUDTException* CUDTUnited::getError()
+CUDTException& CUDTUnited::getError()
 {
-    if(!pthread_getspecific(m_TLSError))
-        pthread_setspecific(m_TLSError, new CUDTException);
-    return (CUDTException*)pthread_getspecific(m_TLSError);
+    if (!GetThreadLocalError())
```

Comment:
This logic can be hidden inside `GetThreadLocalError`. This way you can also decide inside the particular platform implementation whether you need this lazy initialization or not.